aphrodite/modeling/layers/sampler.py

```diff
@@ -508,24 +509,45 @@ def _apply_epsilon_cutoff(
 def _apply_typical_sampling(
     logits: torch.Tensor,
     typical_p: torch.Tensor,
+    typical_p_sigma: torch.Tensor,
 ) -> torch.Tensor:
-    typ_p = torch.tensor(typical_p, dtype=logits.dtype, device=logits.device)
+    typ_p = typical_p.clone().detach().to(logits.device).to(logits.dtype)
+    typ_threshold = typical_p_sigma.clone().detach().to(logits.device).to(
+        logits.dtype)
+    THRESHOLD = 1000
+
     shifted_logits = torch.log_softmax(logits, dim=-1)
-    probs = shifted_logits.exp()
+    probs = torch.exp(shifted_logits)
 
     neg_entropy = (probs * shifted_logits).nansum(dim=-1, keepdim=True)
+    # NOTE: We don't take the absolute value of the surprisal deviations
+    # This deviates from the original implementation
+    surprisal_deviations = neg_entropy - shifted_logits
```

Comment:
I disagree with this change (or the intentions here). The modification in my original hack (not posted here) was intended to retain the basic behavior of Typical_P, which first sorts the surprisal deviations by their absolute value.

Only after this is done, then, using the signed surprisal deviations (copied into a different tensor before computing the absolute value for the other), you would obtain a second subset for extending the token selection as in the algorithm described in the explanation posted in the PR discussion.